tests/aiohttp/test_get_response.py

```diff
@@ -0,0 +1,150 @@
+from connexion.apis.aiohttp_api import AioHttpApi
+from connexion.lifecycle import ConnexionResponse
+
+
+@pytest.fixture(scope='module')
+def api(aiohttp_api_spec_dir):
+    yield AioHttpApi(specification=aiohttp_api_spec_dir / 'swagger_secure.yaml')
+
+
+@asyncio.coroutine
+def test_get_response_from_aiohttp_response(api):
+    response = yield from api.get_response(web.Response(text='foo', status=201, headers={'X-header': 'value'}))
+    assert isinstance(response, web.Response)
+    assert response.status == 201
+    assert response.body == b'foo'
+    assert response.content_type == 'text/plain'
+    assert dict(response.headers) == {'Content-Type': 'text/plain; charset=utf-8', 'X-header': 'value'}
+
+
+@asyncio.coroutine
+def test_get_response_from_connexion_response(api):
+    response = yield from api.get_response(ConnexionResponse(status_code=201, mimetype='text/plain', body='foo', headers={'X-header': 'value'}))
+    assert isinstance(response, web.Response)
+    assert response.status == 201
+    assert response.body == b'foo'
+    assert response.content_type == 'text/plain'
+    assert dict(response.headers) == {'Content-Type': 'text/plain; charset=utf-8', 'X-header': 'value'}
+
+
+@asyncio.coroutine
+def test_get_response_from_string(api):
+    response = yield from api.get_response('foo')
+    assert isinstance(response, web.Response)
+    assert response.status == 200
+    assert response.body == b'foo'
+    assert response.content_type == 'text/plain'
+    assert dict(response.headers) == {'Content-Type': 'text/plain; charset=utf-8'}
+
+
+@asyncio.coroutine
+def test_get_response_from_string_status(api):
+    response = yield from api.get_response(('foo', 201))
+    assert isinstance(response, web.Response)
+    assert response.status == 201
+    assert response.body == b'foo'
+    assert response.content_type == 'text/plain'
+    assert dict(response.headers) == {'Content-Type': 'text/plain; charset=utf-8'}
+
+
+@asyncio.coroutine
+def test_get_response_from_string_status_headers(api):
+    response = yield from api.get_response(('foo', 201, {'X-header': 'value'}))
+    assert isinstance(response, web.Response)
+    assert response.status == 201
+    assert response.body == b'foo'
+    assert response.content_type == 'text/plain'
+    assert dict(response.headers) == {'Content-Type': 'text/plain; charset=utf-8', 'X-header': 'value'}
+
+
+@asyncio.coroutine
+def test_get_response_from_tuple_error(api):
+    with pytest.raises(TypeError) as e:
+        yield from api.get_response((web.Response(text='foo', status=201, headers={'X-header': 'value'}), 200))
+    assert str(e.value) == "Cannot return web.StreamResponse in tuple. Only raw data can be returned in tuple."
+
+
+@asyncio.coroutine
+def test_get_response_from_dict(api):
+    response = yield from api.get_response({'foo': 'bar'})
+    assert isinstance(response, web.Response)
+    assert response.status == 200
+    assert response.body == b"{'foo': 'bar'}"
```

Comment:
You wouldn't expect this to be json?

Reply:
Right. I would not expect this to be json. Train of thought follows:
- Everything in this file is testing against an api that has only one endpoint `/greeting/{name}` -> `fakeapi.aiohttp_handlers.aiohttp_post_greeting` which returns a dict.
- But it's not actually using the handler, it's accessing get_response directly.
- So mimetype is not coming from the spec here, which means there is no assumption that the content will be json.
- which means if you pass an object (in this case a dict) it will be stringified by `_jsonify_data`.
- the implementation of `_jsonify_data` in `abstract` is basically doing the same thing that `_cast_body` did in `aiohttp_api`
- Note that this body is using `'` not `"` so it is not valid json. it is just a stringified dict.
- So, yes I would expect (without changing behavior) this to merely stringify the dict as no mimetype / content type is defined.
- Consider the next test where a mimetype is explicitly defined, and therefore produces json data.

Reply:
I think I can follow your logic here - I have a minor pragmatic hangup which is that casting a dict to a string is kind of the least-useful way to serialize that object.
You might argue in practice that folks would notice this, and then include the right content-type, but it seems like an unfriendly default.
Consider the following flask app, which defaults to returning json
```
  import flask                                                                                                                                                                                                      
      
  app = flask.Flask(__name__)    
      
  @app.route("/")    
  def a():    
      return {"a": "b"}    
      
  app.run()
```
Does that mean in that example you'd expect to get a string representation of the dict as `text/plain`?